### DISCUSSION TITLE
ANTS and Iris Coord System..

### DISCUSSION THREAD

**Discussion by raghu330** [SciTools/iris#5154] [Q&A] [answered]
Hello All!

This is more to do with ANTS which uses Iris, but my problem is basic coord issue. 

I would like to convert (regrid) an ancillary file(s) from 0.25 deg resolution to ~0.125 deg to run JULES land surface model at 12 km. My meteorological forcings too at this higher resolution. I have taken the land-sea mask at ~0.125 deg as the target grid like:

>>> target_grid = ants.load(target_file)[0]   ## Output is an iris cube

I have the source ancillary data (in nc file format @ 0.25 deg) containing all the required ancillaries to drive JULES and I load them in ANTS as an iris cube.

>>> source_cubes = ants.load(source_ancil_file)

Now I am defining the regrid scheme as:
>>> horizontal_scheme = ants.regrid.rectilinear.Linear()
>>> scheme = ants.regrid.GeneralRegridScheme(horizontal_scheme = horizontal_scheme)

Regrid the source ancillaries to new target resolution
>>> new_lsmask = source_cubes[25].regrid(target_grid, scheme)

I get the following error when I try to regrid:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/raghav/anaconda3/envs/ants/lib/python3.7/site-packages/iris/cube.py", line 4560, in regrid
    regridder = scheme.regridder(self, grid)
  File "/home/raghav/ANTS/ants.cur/lib/ants/regrid/__init__.py", line 149, in regridder
    hregridder = self._horizontal_scheme.regridder(src_grid, target_grid)
  File "/home/raghav/ANTS/ants.cur/lib/ants/regrid/rectilinear.py", line 549, in regridder
    src_grid, target_grid, "linear", self._extrapolation
  File "/home/raghav/ANTS/ants.cur/lib/ants/regrid/rectilinear.py", line 464, in __init__
    args[0] = _process_cube_crs(args[0])
  File "/home/raghav/ANTS/ants.cur/lib/ants/regrid/rectilinear.py", line 31, in _process_cube_crs
    src_crs = sx.coord_system.as_ants_crs()
AttributeError: 'NoneType' object has no attribute 'as_ants_crs'

Now how can I set the coordinate system for both the target and source grid to say "GeogCS(6371229.0)" or something meaningful to regrid the source data to target grid size?

Thank you & Regards
Raghu

**Comment by raghu330** [comment]
Hello All!

I was able to use just Iris instead of ANTS to get the data regridded. Hope JULES accepts these ancillaries!

Thanks & Regards
Raghu

**Comment by trexfeathers** [comment, marked as answer]
>Now how can I set the coordinate system for both the target and source grid to say "GeogCS(6371229.0)" or something meaningful to regrid the source data to target grid size?

```python
from iris.coord_systems import GeogCS

for axis in ("x", "y"):
    coord = my_cube.coord(axis=axis)
    coord.coord_system = GeogCS(6371229.0)
```

**Reply by raghu330** [reply]
Thanks Martin!
